internal/backends/mcu/lengths.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
```

Comment:
This feels like something that could be in the core library, so that we could have `geometry()` on the `Item` trait for example return a `LogicalRect`. Or the compiler could generate `PhysicalLength` for embedded glyph metrics. The other backends would probably benefit as well.

The main issue I see is that PhysicalLength should be f32 for GL/Qt and i16 for the mcu.